apps/src/code-studio/assets/SpriteUpload.js

```diff
@@ -1,18 +1,53 @@
 import React from 'react';
 
 export default class SpriteUpload extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      file: null
-    };
-  }
+  state = {
+    fileData: null,
+    filePreviewURL: '',
+    filename: '',
+    category: ''
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+
+    return fetch(
+      `/api/v1/animation-library/level_animations/${this.state.filename}`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'image/png'
+        },
+        body: this.state.fileData
+      }
+    )
+      .then(response => {
+        let responseText = document.getElementById('sprite-upload-error');
```

Comment:
this (and in `catch` below) is mixing imperative DOM manipulation with react -- let's just use react here. you'll want to set the error in component state, then render that state in the `<p>` below. along with that, we should clear the error on change (e.g., there's an error, then a new file is uploaded or the category is changed -- we'll want to make sure the original error doesn't persist)